File: Level 2/구명보트/solution.js
function solution(people, limit) {
  let answer = 0;
  people = people.sort((a, b) => b - a);

  let leftPersonIndex = 0;
  let rightPersonIndex = people.length - 1;

  while (leftPersonIndex <= rightPersonIndex) {
    const sumWeight = people[leftPersonIndex] + people[rightPersonIndex];
    if (sumWeight > limit) {
      leftPersonIndex++;
    } else {
      leftPersonIndex++;
      rightPersonIndex--;
    }
    answer++;
  }

  return answer;
}

File: Level 2/구명보트/solution.py
def solution(people, limit):
    people.sort()

    answer = 0

    i = 0
    j = len(people) - 1

    while i <= j:
        if people[i] + people[j] <= limit:
            i += 1
            j -= 1
        else:
            j -= 1
        answer += 1

    return answer
